http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/ProxyConfig.scala

```diff
@@ -9,4 +11,9 @@ case class ProxyConfig(proxyDomain: String,
                        tokenCacheExpiryTime: FiniteDuration,
                        tokenCacheMaxSize: Int,
                        internalIdCacheExpiryTime: FiniteDuration,
-                       internalIdCacheMaxSize: Int)
+                       internalIdCacheMaxSize: Int) {
+  def getProxyServerHostName: String = {
+    val url = new URL(proxyUrlBase)
+    s"${url.getProtocol}://${url.getHost}" + (if (url.getPort == -1) "" else s":${url.getPort.toString}")
```

Comment:
when would it be `-1`?

Reply:
From javadoc:
```
    /**
     * Gets the port number of this {@code URL}.
     *
     * @return  the port number, or -1 if the port is not set
     */
```

It's `-1` if the URL string is `https://notebooks.firecloud.org`. But it's set if the URL is `https://leonardo-fiab.dsde-dev.broadinstitute.org:30443` (I think fiab is the only time we explicitly set a port).